.github/workflows/tests_windows.yml

```diff
@@ -0,0 +1,52 @@
+name: Tests
+on: [push, pull_request]
+jobs:
+  test-windows:
+    strategy:
+      fail-fast: false
+      matrix:
+        target:
+          - windows-amd64-unit-test-4-cpu
+        # FIXME(fuweid):
+        #
+        # The windows will throws the following error when enable race.
+        # We skip it until we have solution.
+        #
+        #   ThreadSanitizer failed to allocate 0x000200000000 (8589934592) bytes at 0x0400c0000000 (error code: 1455)
+        #
+        #- windows-amd64-unit-test-4-cpu-race
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-go@v4
+        with:
+          go-version: "1.21.1"
+      - run: make fmt
+      - env:
+          TARGET: ${{ matrix.target }}
+        run: |
+          case "${TARGET}" in
+            windows-amd64-unit-test-4-cpu)
+              CPU=4 make test
+              ;;
+            *)
+              echo "Failed to find target"
+              exit 1
+              ;;
+          esac
+        shell: bash
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@3a919529898de77ec3da873e3063ca4b10e7f5cc # v3.7.0
+
+  coverage:
+    needs: ["test-windows" ]
+    strategy:
+      matrix:
+        os: [windows-latest ]
```

Comment:
Please add the coverage for ubuntu-latest in tests_amd64.yaml.

Reply:
it is in the template, it will inherited 🤔 

ok will inline it now

Reply:
i still think it is better to be inherited from the template, wdyt ?

Reply:
I didn't want to run coverage on arm64, but do not have strong reason. 

I am not sure you correctly configured the field `needs` (line 67 in the template file), leave it to @fuweid to take a closer look.